README.rst

```diff
@@ -26,11 +26,100 @@ View management for Amazon's Redshift
 * Free software: Apache Software License 2.0
 * Documentation: https://room-with-a-view.readthedocs.io.
 
-
 Features
 --------
 
-* TODO
+Room with a view is a python script that automatically parses a collection of
+SQL files to find View and Function definitions and their dependencies. It then
+makes it easy to sync those views with Redshift, automatically dropping and
+recreating dependent views as necessary so that there are no errors. The key
+benefits are:
+
+* No writing code to send view SQL to Redshift: the script does it for you.
+* Edit any view and sync it without worrying about needing to drop or recreate
+  views that depend on it.
+* Error handling tells you exactly where there are errors in your views.
+* The script runs in an atomic transaction, so you can't accidentally enter a
+  broken state by syncing some views and not others.
+
+
+Setup
+-----
+
+* Install the package: ``pip install room_with_a_view``.
+
+* Create ``settings.yaml``, and edit the file to configure your Redshift connection and the location of your .sql files. Example ``settings.yaml`` file:
+
+.. code-block:: yaml
+
+   connections:
+     default:
+       host: localhost
+       port: 5432
+       user: awsuser
+       password: **CHANGEME**
+       dbname: postgres
+
+   directories:
+     default: .
+
+* You're ready to go! Try ``room_with_a_view sync-all`` to sync all your views, or ``room_with_a_view --help`` to learn more about the command.
+
+Usage
+-----
+
+::
+
+    usage: room_with_a_view.py [-h]
+                               [--view-names [VIEW-OR-FUNCTION-NAME [VIEW-OR-FUNCTION-NAME ...]]]
+                               [--file-names [FILE-PATH [FILE-PATH ...]]]
+                               [--connection CONNECTION]
+                               [--directories [DIRECTORY [DIRECTORY ...]]]
+                               [--settings SETTINGS] [--verbosity VERBOSITY]
+                               {sync,drop-all,sync-all,list,drop}
+
+    Manages Redshift SQL views. Possible actions:
+        sync: Syncs specific views or functions (identified by the --view-names or --file-names parameters).
+        drop-all: Drops all views and functions in all .sql files in a set of directories (identified by the --directories parameter). The directory will be searched recursively.
+        sync-all: Syncs all views and functions in all .sql files in a set of directories (identified by the --directories parameter). The directory will be searched recursively.
+        list: lists all known views and functions.
+        drop: Drops specific views or functions (identified by the --view-names or --file-names parameters).
+
+    positional arguments:
+      {sync,drop-all,sync-all,list,drop}
+                            The action to perform.
+
+    optional arguments:
```

Comment:
i don't feel strongly, but i'd be OK w/o the --directories and --file-names args---we can look at the yaml to figure it out, and filter to views/funcs from there

Reply:
`--filenames` serves a purpose: it tells the script to sync functions and views in specific files.

`--directories` names places to look for SQL to use to build the dependency graph. I eliminated it and converted the yaml to a list, as you suggest.